onnxconverter_common/auto_mixed_precision_model_path.py

```diff
@@ -0,0 +1,247 @@
+
+    return segments.get_nodes()
+
+
+def generate_temp_filename(target_model_path):
+    target_model_folder = os.path.dirname(target_model_path)
+    if not os.path.exists(target_model_folder):
+        os.mkdir(target_model_folder)
+    tensor_filename = str(uuid.uuid1())
+    onnx_filename = os.path.join(target_model_folder, tensor_filename + ".onnx")
+    return onnx_filename, tensor_filename + ".data"
+
+
+def _copy_fp32_model(**kwargs):
+    source_model_path = kwargs.get('source_model_path')
+    input_feed = kwargs.get('input_feed')
+    providers = kwargs.get('providers')
+    tmp_model32_path = kwargs.get("tmp_model32_path")
+    tmp_model32_tensor_name = kwargs.get("tmp_model32_tensor_name")
+
+    model_32 = onnx.load(source_model_path)
+    save_model(model_32, tmp_model32_path, location=tmp_model32_tensor_name)
+
+    print("infer_shape_path for", tmp_model32_path, tmp_model32_tensor_name)
+    shape_inference.infer_shapes_path(tmp_model32_path)
+    model_32 = onnx.load(tmp_model32_path)
+
+    output_32 = inference(tmp_model32_path, input_feed, providers=providers)
+
+    kwargs["res1"] = output_32
+    kwargs["res2"] = output_32
+    if not _validate_result(**kwargs):
+        raise ValueError("validation failed for fp32 model")
+
+    return model_32, output_32
+
+
+def _validate_result(**kwargs):
+    validate_fn = kwargs.get("validate_fn")
+    rtol = kwargs.get("rtol")
+    res1 = kwargs.get("res1")
+    res2 = kwargs.get("res2")
+
+    if validate_fn is not None and not validate_fn(res1, res2):
+        return False
+    if rtol is not None:
+        for r1, r2 in zip(res1, res2):
+            if not np.allclose(r1, r2, rtol):
+                return False
+    return True
+
+
+def _convert_and_check_inference_result(**kwargs):
+    model_32 = kwargs.get("model_32")
+    keep_io_types = kwargs.get("keep_io_types")
+    is_final_model = kwargs.get("is_final_model")
+    target_model_path = kwargs.get("target_model_path")
+    node_block_list = kwargs.get("node_block_list")
+    input_feed = kwargs.get("input_feed")
+    providers = kwargs.get("providers")
+    tmp_model32_tensor_name = kwargs.get("tmp_model32_tensor_name")
+
+    _print_node_block_list(node_block_list)
+    # convert to fp16
+    model_16 = float16.convert_float_to_float16(
+        copy.deepcopy(model_32), node_block_list=node_block_list,
+        keep_io_types=keep_io_types, disable_shape_infer=True)
+    # need to save model to model_path here.....
+    if not is_final_model:
+        location = tmp_model32_tensor_name  # using temporary file name
+    else:
+        location = kwargs.get("location")  # using the speficified external data file name
+    save_model(model_16, target_model_path, location=location)
+    # inference
+    output_16 = inference(target_model_path, input_feed, providers=providers)
+    kwargs["res2"] = output_16
+    result = _validate_result(**kwargs)
+    print("validate result = ", result)
```

Comment:
Printing this comment is not necessary because we will check the returned validation results of this function if we need. And this might be output a lot of times if we have lots of nodes. So, let's remove it.